packages/core/src/mpi/match-patients.ts

```diff
@@ -315,23 +315,71 @@ export function strictMatchingAlgorithm(patient1: PatientData, patient2: Patient
   const isDobMatch = normalizedPatient1.dob === normalizedPatient2.dob;
   const isGenderMatch = normalizedPatient1.genderAtBirth === normalizedPatient2.genderAtBirth;
 
-  const firstNames1 = splitName(normalizedPatient1.firstName);
-  const firstNames2 = splitName(normalizedPatient2.firstName);
+  const cleanedFirstName1 = removeCommonPrefixesAndSuffixes(normalizedPatient1.firstName);
+  const cleanedFirstName2 = removeCommonPrefixesAndSuffixes(normalizedPatient2.firstName);
 
-  const lastNames1 = splitName(normalizedPatient1.lastName);
-  const lastNames2 = splitName(normalizedPatient2.lastName);
+  const cleanedLastName1 = removeCommonPrefixesAndSuffixes(normalizedPatient1.lastName);
+  const cleanedLastName2 = removeCommonPrefixesAndSuffixes(normalizedPatient2.lastName);
 
-  const hasMatchingFirstName =
-    firstNames1.length === firstNames2.length &&
-    firstNames1.every(name => firstNames2.includes(name));
-  const hasMatchingLastName =
-    lastNames1.length === lastNames2.length && lastNames1.every(name => lastNames2.includes(name));
+  const firstNames1 = splitName(cleanedFirstName1);
+  const firstNames2 = splitName(cleanedFirstName2);
+
+  const lastNames1 = splitName(cleanedLastName1);
+  const lastNames2 = splitName(cleanedLastName2);
+
+  const hasMatchingFirstName = firstNames1.some(name => firstNames2.includes(name));
+  const hasMatchingLastName = lastNames1.some(name => lastNames2.includes(name));
```

Comment:
Biggest change is here instead of checking everything matches just one needs too